tests/python_package_test/test_dask.py

```diff
@@ -1108,3 +1108,26 @@ def test_dask_methods_and_sklearn_equivalents_have_similar_signatures(methods):
     for param in dask_spec.args:
         error_msg = f"param '{param}' has different default values in the methods"
         assert dask_params[param].default == sklearn_params[param].default, error_msg
+
+
+def test_training_succeeds_when_data_is_dataframe_and_label_is_column_array(
```

Comment:
can you please run this test for all three of `DaskLGBMClassifier`, `DaskLGBMRanker`, and `DaskLGBMRegressor`? Similar to https://github.com/microsoft/LightGBM/blob/84c4b7504a49eed63fa819a92db7d0b18292f3f4/tests/python_package_test/test_dask.py#L966

Reply:
This only seems to impact the regressor, the ranker and the classifier go through scikit-learns check_X_y

Reply:
testing all three will prevent us from introducing a change in the future that makes that not true

Reply:
Yeah that sounds reasonable. I'll include it in test_sklearn as well

Reply:
Is there a way to make the model deterministic? I'm calling repeatedly `model_1d.fit(dX, dy, sample_weight=dw).predict(dX).compute()` and the results differ from time to time. I've set `random_state` in the parameters.

Reply:
you can try adding "deterministic: true" to the parameters: https://lightgbm.readthedocs.io/en/latest/Parameters.html#deterministic

Reply:
I had to switch to single worker training for the test to pass everytime

Reply:
ok, I'd prefer removing the `assert_eq()` then and going back to multi-worker training, sorry. It can be difficult to guarantee consistent results when doing distributed training on such tiny data sizes. I had hoped it would work but if it doesn't, we can go back to just training one model and checking `.fitted_`. Sorry for sending you down a road to a dead end.

Reply:
Sorry hadn't seen this. Even though the training is only happening in one worker the idea is testing that both the dask and scikit-learn models can train with a column array and get the same results, isn't it?

Reply:
For the Dask interface, having data in a single partition is basically just falling back to training with the sklearn estimators.

The test you've added in `test_sklearn.py` that ends in `assert np.allclose(preds_1d, preds_2d)` is already checking that, for non-distributed training with the sklearn interface, the model output is the same when a label has shape `[n_samples, 1]` or `[n_samples]`.

Given that, I'd rather have the Dask test be a smoke test that multi-worker distributed training works when data has a column-array shape.